packages/google_sign_in/google_sign_in_web/lib/src/button_configuration.dart

```diff
@@ -0,0 +1,202 @@
+  }
+  return js_util.jsify(<String, Object?>{
+    if (config.type != null) 'type': _idType[config.type],
+    if (config.theme != null) 'theme': _idTheme[config.theme],
+    if (config.size != null) 'size': _idSize[config.size],
+    if (config.text != null) 'text': _idText[config.text],
+    if (config.shape != null) 'shape': _idShape[config.shape],
+    if (config.logoAlignment != null)
+      'logo_alignment': _idLogoAlignment[config.logoAlignment],
+    if (config.minimumWidth != null) 'width': config.minimumWidth,
+    if (config.locale != null) 'locale': config.locale,
+  }) as id.GsiButtonConfiguration;
+}
+
+/// A class to configure the Google Sign-In Button for web.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#GsiButtonConfiguration
+class GSIButtonConfiguration {
+  /// Constructs a button configuration object.
+  GSIButtonConfiguration({
+    this.type,
+    this.theme,
+    this.size,
+    this.text,
+    this.shape,
+    this.logoAlignment,
+    this.minimumWidth,
+    this.locale,
+  }) : assert(minimumWidth == null || minimumWidth > 0);
+
+  /// The button type: icon, or standard button.
+  final GSIButtonType? type;
+
+  /// The button theme.
+  ///
+  /// For example, filledBlue or filledBlack.
+  final GSIButtonTheme? theme;
+
+  /// The button size.
+  ///
+  /// For example, small or large.
+  final GSIButtonSize? size;
+
+  /// The button text.
+  ///
+  /// For example "Sign in with Google" or "Sign up with Google".
+  final GSIButtonText? text;
+
+  /// The button shape.
+  ///
+  /// For example, rectangular or circular.
+  final GSIButtonShape? shape;
+
+  /// The Google logo alignment: left or center.
+  final GSIButtonLogoAlignment? logoAlignment;
+
+  /// The minimum button width, in pixels.
+  ///
+  /// The maximum width is 400 pixels.
+  final double? minimumWidth;
+
+  /// The pre-set locale of the button text.
+  ///
+  /// If not set, the browser's default locale or the Google session user's
+  /// preference is used.
+  ///
+  /// Different users might see different versions of localized buttons, possibly
+  /// with different sizes.
+  final String? locale;
+}
+
+/// The type of button to be rendered.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#type
+enum GSIButtonType {
+  /// A button with text or personalized information.
+  standard,
+
+  /// An icon button without text.
+  icon;
+}
+
+const Map<GSIButtonType, id.ButtonType> _idType =
+    <GSIButtonType, id.ButtonType>{
+  GSIButtonType.icon: id.ButtonType.icon,
+  GSIButtonType.standard: id.ButtonType.standard,
+};
+
+/// The theme of the button to be rendered.
+///
+/// https://developers.google.com/identity/gsi/web/reference/js-reference#theme
+enum GSIButtonTheme {
+  /// A standard button theme.
+  outline,
+
+  /// A blue-filled button theme.
+  filledBlue,
+
+  /// A black-filled button theme.
+  filledBlack;
+}
+
+const Map<GSIButtonTheme, id.ButtonTheme> _idTheme =
```

Comment:
These themes are also given by the [button configuration](https://developers.google.com/identity/gsi/web/reference/js-reference?hl=de#theme), so it looks like I can only use these options? My hope was that we can change the color at least to our app default button color. 

The JavaScript API seems quite restrictive here. I guess this can be a heavy design breaker for many "login tower" pages. Any chance to get this changed within the JavaScript API?

Reply:
(I couldn't find your specific report yet, but believe me, you're not alone in asking for extra customization options!)

Reply:
:heart: thank you :), if it helps I remember I linked both the github issue and the PR within the feedback message.

Reply:
I couldn't find it internally, but I'll keep my eyes peeled. As I said, there's a lot of people asking for additional customization options. We'll just make them available through the plugin as they get released!